library/ssl_tls12_server.c

```diff
@@ -3946,7 +3946,7 @@ static int ssl_parse_client_key_exchange(mbedtls_ssl_context *ssl)
            The size of the array is thus greater than 256 bytes which is greater than any
            possible value of ecpoint_len (type uint8_t) and the check below can be skipped.*/
 #if !defined(PSA_WANT_ALG_FFDH)
-        if (ecpoint_len > sizeof(handshake->xxdh_psa_peerkey)) {
+        if ((size_t) ecpoint_len > sizeof(handshake->xxdh_psa_peerkey)) {
```

Comment:
There is a similar piece of code in `ssl_parse_server_ecdh_params` in `ssl_tls12_client.c` which had the same problem. We fixed it in 7910cdd47fab98e29af6a1c6c7a4730f4ece8a58 by making `ecpoint_len` have the type `size_t`. This was needed for GCC, where casting `ecpoint_len` in the comparison expression isn't enough to disable the warning. So I think we should do the same here.

Reply:
Good point, I should have done some digging there. It would have been too easy ^^ Done in 586dcd9